### ISSUE TITLE
encoding.TextUnmarshaler for slice type

### ISSUE THREAD

**Issue by hsinhoyeh** [gorilla/schema#62]
Hi,
thanks for creating such great package. It did save my time.

For my understanding, if a type implemented encoding.TextUnmarshal interface, it should be
decode according to how we implement UnmarshalText function, right?
But slice type is a different story, because you miss test the interface in L137 https://github.com/gorilla/schema/blob/master/decoder.go#L137

and the following changes works for me

```
+        _, ok := v.Interface().(encoding.TextUnmarshaler)
-       if conv == nil && t.Kind() == reflect.Slice {
+       if conv == nil && t.Kind() == reflect.Slice && !ok {
                var items []reflect.Value
                elemT := t.Elem()
                isPtrElem := elemT.Kind() == reflect.Ptr
                if isPtrElem {
                        elemT = elemT.Elem()
                }
```

let me know my understand is wrong, thanks

**Comment by elithrar** [comment]
This relates to https://github.com/gorilla/schema/pull/58

**Comment by kisielk** [comment]
I believe this is fixed by #84
